tests/python/test_dataset_methods.py

```diff
@@ -27,6 +27,19 @@
 import numpy as np
 from smartredis import Dataset
 
+def test_serialize_dataset():
+    """Test serializing a dataset
+    """
+    dataset = Dataset("test-dataset")
+    data = np.uint8([2,4,8])
+    dataset.add_tensor("u8_tensor", data)
+    data = np.double([2.0,4.1,8.3, 5.6])
+    dataset.add_tensor("double_tensor", data)
+    dataset.add_meta_scalar("float2_scalar", float(3.1415926535))
+    dataset.add_meta_scalar("float_scalar", np.double(3.1415926535))
+    dataset.add_meta_string("metastring", "metavalue")
+    print(dataset)
```

Comment:
Little worried about this test: technically speaking, w/o the changes in this PR, this should still pass.

Maybe just check to make sure that the string of a dataset is no equal to the repr (which is the default when casting an object to a str)

```py
dataset = Dataset(...)
...
assert repr(dataset) != str(dataset)
```

Or you could search for a sub-string that we expect to be in the "string-ified" dataset

```py
assert "Dataset(test-dataset):" in str(dataset)
```

Reply:
Done